### ISSUE TITLE
some problems in bitextor-align-documents.py【not sure, but I think my logic is right】

### ISSUE THREAD

**Issue by JiaYueHuang** [bitextor/bitextor#69]
in line 111：
candidateIterations.append(num_candidates-1)
should be replaced by：
candidateIterations.append(num_candidates)

otherwise, if the first optimal candidate document is paired, the second candidate (suboptimal candidate, if exist) always be ignored.

**Comment by lpla** [comment]
Hello again @JiaYueHuang . I am quite sure that line shouldn't be replaced as the previous line does a `num_candidates = num_candidates+1`

Also, the behavior you are describing is not managed in that line, or "if" block. Two lines before there is a comment after the code line:

```
elif int(field_n[0]) in pairedDocs and num_candidates < (len(fields)-1): #If the documents are already paired, we just ignored the read candidate and do another iteration
```
Which means that if the document is already in the paired list (that means, after N iterations looking for symmetrical paired documents, by default), do one more iteration looking for candidates to compensate (if there are more candidates in the idx file). Exactly the same code in line 131.

Take into account that this code will be only triggered in case you are using the `--iterations` option, given that `pairedDocs` will be always empty at the first iteration document pairing and filled after checking candidate relations/pairing in both directions.

So, I think that the behavior you are describing in the issue is managed in lines 101-108, which create a list of `best_ridx2` which are the N best candidates of each document in ridx2 direction. Same code in lines 123-130, for ridx1 direction. After that, we pair symmetric (or asymmetric pairs if using `--nonsymmetric` option) candidate alignments, filling `pairedDocs`, which will be taken into account if using `--iterations` options to ignore paired documents, as I said before.

I know that this ancient code is hard to follow (and I usually need some time to understand it after some time without working with it), but I think that it is working as intended.
